### PR TITLE
Various bug fixes

### PR DESCRIPTION
**Pull request by zacikpa**
This PR fixes 3 bugs I encountered when analyzing cryptographic libraries. For details, see commit messages.

### CODE REVIEW

File: diffkemp/output.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def _create_definitions(self):
     def _create_function_defs(self):
         definitions = {}
         for name in self.function_names:
-            if name not in self.result.graph.vertices:
+            if not self.result.graph or name not in self.result.graph.vertices:
```
Comment (viktormalik) [inline diffkemp/output.py:84]: Move the new check out of the loop.

File: diffkemp/output.py [hunk 2]
```diff
@@ -109,7 +109,8 @@ def _create_type_defs(self):
         """
         definitions = {}
         for type_name, parent_name in self.type_names:
-            if parent_name not in self.result.graph.vertices:
+            if not self.result.graph or \
```
Comment (viktormalik) [inline diffkemp/output.py:112]: Same as above.

File: diffkemp/simpll/passes/CalledFunctionsAnalysis.h [hunk 3]
```diff
@@ -38,7 +38,9 @@ class CalledFunctionsAnalysis
     void collectCalled(const Function *Fun, Result &Called);
     /// Looks for functions in a value (either a function itself, or a composite
     /// type constant).
-    void processValue(const Value *Val, Result &Called);
+    void processValue(const Value *Val,
+                      Result &Called,
+                      std::set<const Value *> &Processed);
```
Comment (viktormalik) [inline diffkemp/simpll/passes/CalledFunctionsAnalysis.h:43]: Could we make `Processed` a private class member?

**Review by viktormalik — CHANGES_REQUESTED** [review]

**Comment by PLukas2018** [comment]
Thanks for the check if `graph` is not `None`, I did not realise it could happen. I think the better place for the check would be before creating the definitions,
https://github.com/viktormalik/diffkemp/blob/f1e801dabbe9c9c35052c7900f1b9725bb230519/diffkemp/output.py#L28-L34
because both of the methods uses the `graph`.

**Comment by viktormalik** [comment]
> Thanks for the check if `graph` is not `None`, I did not realise it could happen. I think the better place for the check would be before creating the definitions,
> 
> https://github.com/viktormalik/diffkemp/blob/f1e801dabbe9c9c35052c7900f1b9725bb230519/diffkemp/output.py#L28-L34
> 
> 
> because both of the methods uses the `graph`.

I'd argue that it's up to the methods to check that everything they need exists. If we added the check to the caller method, we'd have to add checks for graph existence for all future calls of `_create_results` and `_create_definitions`. I'm aware that these are class-internal methods which are very unlikely to be called from another place, but it's safer (although slightly more verbose) this way.

**Comment by PLukas2018** [comment]
> I'd argue that it's up to the methods to check that everything they need exists. If we added the check to the caller method, we'd have to add checks for graph existence for all future calls of `_create_results` and `_create_definitions`. I'm aware that these are class-internal methods which are very unlikely to be called from another place, but it's safer (although slightly more verbose) this way.

Thanks, that makes sense.

File: diffkemp/output.py (hunk 1 above)
Comment (zacikpa) [reply diffkemp/output.py]: Done.

File: diffkemp/output.py (hunk 2 above)
Comment (zacikpa) [reply diffkemp/output.py]: Done.

File: diffkemp/simpll/passes/CalledFunctionsAnalysis.h (hunk 3 above)
Comment (zacikpa) [reply diffkemp/simpll/passes/CalledFunctionsAnalysis.h]: We certainly can. That's a much cleaner solution, thank you.

**Review by viktormalik — APPROVED** [review]
LGTM, thanks!

The CI failure doesn't seem to come from this PR, I deleted the CI caches and restarted it.

Can you squash the commits, please? 2 should be sufficient (1 for YAML fix and 1 for recursion fix).

**Comment by zacikpa** [comment]
Done, thanks for the review.

**Pull request merged**